classification.py

```diff
@@ -47,7 +47,10 @@ def classify(self, features):
 
 class KNNClassifier(Classifier):
     def __init__(self, k=1, debug=False):
-        self.knn = cv2.KNearest()
+        if cv2.__version__.split(".")[0] == "3":
```

Comment:
Can you extract the version detection logic to a function, so we avoid duplicating logic? Something like `is_opencv_v3() == True` in `opencv_utils.py`? Or maybe `get_opencv_version() == 3`